lib/msf/core/post/windows/services.rb

```diff
@@ -181,32 +183,61 @@ def each_service(&block)
   # @todo Rewrite to allow operating on a remote host
   #
   def service_list
-    if load_extapi
-      return session.extapi.service.enumerate
-    else
-      serviceskey = "HKLM\\SYSTEM\\CurrentControlSet\\Services"
-      a =[]
-      services = []
-      keys = registry_enumkeys(serviceskey)
-      keys.each do |s|
-        if a.length >= 10
-          a.first.join
-          a.delete_if {|x| not x.alive?}
+    SERVICES.clear
+    case session.type
+    when "powershell"
+      pscommand = '$services = Get-WmiObject win32_service | ?{$_} | where {($_.pathname -ne $null)}'
```

Comment:
Why are you not using Get-Service native powershell command? That would allow us to retrieve the correct Displayname etc